### DISCUSSION TITLE
Display not updated and not returning from "Printing" screen

### DISCUSSION THREAD

**Discussion by recliq** [OpenNeptune3D/OpenNept4une#75] [Q&A] [answered]
First of all a big thank you for creating and maintaining this great project!
Since I installed this firmware my Neptune 4 Pro is so much more usable and reliable.

I know the display support is a work in progress and still in BETA phase, but I just wanted to know if the following is a common issue or
special to my setup...

All in all the display already works quite well, but i have one issue on my setup:
When I start a print it chages to the "Printing" page as intended and all values are updated. So far so good, but when the KAMP bed leveling starts the display is no longer updaed (eg. temps) and when the print is finished it doesn't come up with the re-print dialog nor returns to main screen. It just hangs in the "Printing" page without updating any values.

**Comment by phillipthelen** [comment]
I have not heard of people running into this before. Can you check the display_connector.log file? It sounds like the script is running into an error that causes it to then not update anymore. The log should hopefully have the error message.
Does the Kamp screen show or is it stuck on the printing screen?

**Comment by recliq** [comment]
It's just stuck in printing screen as soon as KAMP starts...
I will have a look at the log and post any errors...

**Comment by recliq** [comment]
I don't get anything in the log. Had some older entries about incompatible UI firmware although my display firmware is 1.2.11, this is also reported in the display if it works.
Deleted the logfile, rebooted printer (power toggle), the logfile gets recreated but is empty since, although i started 2 prints.
Also tried restarting OpenNept4une.service which didn't do anything.

Can I enbale some kind of debug log?

Edit:
I made another print today and this is the only entry I got during printer start...
`2024-02-05 17:01:54,410 - ERROR - KeyError encountered in software_version_response. Attempting to reconnect.`

**Reply by steve8x8** [reply]
This is related to a moonraker request to return "printer.info" and the subsequent attempt to extract "software_version" from the result (`display.py#L757`). You might add some `logger.info(...)` before the extraction to see what's in the `software_version_response` result. I'm not into Python that much so no suggestions from me what to drop in for the `...` part :/ any suggestions?

**Reply by phillipthelen** [reply]
That is strange that moonraker is not returning the info. Although looking at the code this should not prevent the display from updating. I will look into this and add better handling. The software version isn't critical for the display, so it should be able to handle working without it

**Reply by recliq** [reply]
Just to clarify, that message appeared before the print was started and the display was working fine at that point.
No messages appeared when the display stopped being updated.

**Comment by chuckkay** [comment]
same issue and and other anomolies, is there a command i could ssh or console to restart the elegoo lcd service, to give it a kick in the ass, without killing the print job and restarting..

**Reply by halfmanbear** [reply]
in fluidd the top right (...) menu if you open that then go to services you should be able to restart the service OpenNept4une

**Comment by recliq** [comment]
Ok, I think I finally got at least an error message for you...
This comes from /var/log/syslog as soon as the KAMP leveling starts and the display stops working:
```
2024-02-06T18:32:19.897662+01:00 znp-k1 python[2220]: Task exception was never retrieved
2024-02-06T18:32:19.898536+01:00 znp-k1 python[2220]: future: <Task finished name='Task-7' coro=<DisplayController._process_stream() done, defined at /home/mks/OpenNept4une/display/display.py:835> exception=ValueError("invalid literal for int() with base 10: '(3'")>
2024-02-06T18:32:19.900564+01:00 znp-k1 python[2220]: Traceback (most recent call last):
2024-02-06T18:32:19.901066+01:00 znp-k1 python[2220]:   File "/home/mks/OpenNept4une/display/display.py", line 865, in _process_stream
2024-02-06T18:32:19.901488+01:00 znp-k1 python[2220]:     self.handle_gcode_response(item["params"][0])
2024-02-06T18:32:19.901870+01:00 znp-k1 python[2220]:   File "/home/mks/OpenNept4une/display/display.py", line 1154, in handle_gcode_response
2024-02-06T18:32:19.902857+01:00 znp-k1 python[2220]:     x_count = int(parts[0].strip())
2024-02-06T18:32:19.904032+01:00 znp-k1 python[2220]:               ^^^^^^^^^^^^^^^^^^^^^
2024-02-06T18:32:19.904594+01:00 znp-k1 python[2220]: ValueError: invalid literal for int() with base 10: '(3'
2024-02-06T18:32:19.905644+01:00 znp-k1 python[2220]: Task exception was never retrieved
2024-02-06T18:32:19.907777+01:00 znp-k1 python[2220]: future: <Task finished name='Task-72266' coro=<DisplayController._process_stream() done, defined at /home/mks/OpenNept4une/display/display.py:835> exception=ValueError("invalid literal for int() with base 10: '(3'")>
2024-02-06T18:32:19.908345+01:00 znp-k1 python[2220]: Traceback (most recent call last):
2024-02-06T18:32:19.909467+01:00 znp-k1 python[2220]:   File "/home/mks/OpenNept4une/display/display.py", line 865, in _process_stream
2024-02-06T18:32:19.910310+01:00 znp-k1 python[2220]:     self.handle_gcode_response(item["params"][0])
2024-02-06T18:32:19.910797+01:00 znp-k1 python[2220]:   File "/home/mks/OpenNept4une/display/display.py", line 1154, in handle_gcode_response
2024-02-06T18:32:19.912132+01:00 znp-k1 python[2220]:     x_count = int(parts[0].strip())
2024-02-06T18:32:19.912583+01:00 znp-k1 python[2220]:               ^^^^^^^^^^^^^^^^^^^^^
2024-02-06T18:32:19.913510+01:00 znp-k1 python[2220]: ValueError: invalid literal for int() with base 10: '(3'

```

```
/=======================================================\
|     ~~~~~~~~~~~~~~~~~ [ KIAUH ] ~~~~~~~~~~~~~~~~~     |
|        Klipper Installation And Update Helper         |
|     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~     |
\=======================================================/
/=======================================================\
|     ~~~~~~~~~~~~~~ [ Update Menu ] ~~~~~~~~~~~~~~     |
|-------------------------------------------------------|
| a) [Update all]        |               |              |
|                        | Installed:    | Latest:      |
| Klipper & API:         |---------------|--------------|
|  1) [Klipper]          | v0.12.0-102   | v0.12.0-102  |
|  2) [Moonraker]        | v0.8.0-308    | v0.8.0-309   |
|                        |               |              |
| Klipper Webinterface:  |---------------|--------------|
|  3) [Mainsail]         | v2.9.1        | v2.9.1       |
|  4) [Fluidd]           | v1.27.1       | v1.27.1      |
|                        |               |              |
| Touchscreen GUI:       |---------------|--------------|
|  5) [KlipperScreen]    |               |              |
|                        |               |              |
| Other:                 |---------------|--------------|
|  6) [PrettyGCode]      |               |              |
|  7) [Telegram Bot]     |               |              |
|  8) [Obico for Klipper]|               |              |
|  9) [OctoEverywhere]   |               |              |
| 10) [Mobileraker]      | v0.4.0-42     | v0.4.0-42    |
| 11) [Crowsnest]        | v4.1.3-1      | v4.1.3-1     |
|                        |------------------------------|
```

**Comment by phillipthelen** [comment]
Ok! That is actually very helpful. It seems to be splitting the string about the number of probes it's doing incorrectly. Just need to figure out why. The next time you run KAMP, could you maybe copy paste the line about the number of probes from the console in fluidd? That would be very helpful!

**Reply by recliq** [reply]
I hope this helps, it's the start of a print of the same file:
```
20:14:20  // Found 1 objects
20:14:20  // Original mesh bounds: ((10.0, 21.0),(210.0, 211.0))
20:14:20  // Original probe count: (11,11)
20:14:20  // Adapted mesh bounds: ([105.268, 107.816],[129.732, 127.184])
20:14:20  // Ratio: (0.12232, 0.10193684210526313)
20:14:20  // Adapted probe count: (3,3)
20:14:20  // Generating new points...
20:14:20  // bed_mesh: generated points
// Index | Tool Adjusted | Probe
20:14:20  // 0 | (129.5, 87.4) | (105.3, 107.8)
20:14:20  // 1 | (141.7, 87.4) | (117.5, 107.8)
20:14:20  // 2 | (154.0, 87.4) | (129.7, 107.8)
20:14:20  // 3 | (154.0, 97.0) | (129.7, 117.5)
20:14:20  // 4 | (141.7, 97.0) | (117.5, 117.5)
20:14:20  // 5 | (129.5, 97.0) | (105.3, 117.5)
20:14:20  // 6 | (129.5, 106.7) | (105.3, 127.2)
20:14:20  // 7 | (141.7, 106.7) | (117.5, 127.2)
20:14:20  // 8 | (154.0, 106.7) | (129.7, 127.2)
20:14:21  // probe at 129.513,87.363 is z=0.790975
20:14:22  // probe at 129.513,87.363 is z=0.797225
20:14:24  // probe at 141.750,87.363 is z=0.778427
20:14:25  // probe at 141.750,87.363 is z=0.777177
20:14:26  // probe at 153.975,87.363 is z=0.765879
20:14:27  // probe at 153.975,87.363 is z=0.772129
20:14:29  // probe at 153.975,97.050 is z=0.785879
20:14:30  // probe at 153.975,97.050 is z=0.789629
20:14:31  // probe at 141.750,97.050 is z=0.812177
20:14:32  // probe at 141.750,97.050 is z=0.814677
20:14:34  // probe at 129.513,97.050 is z=0.827225
20:14:35  // probe at 129.513,97.050 is z=0.828475
20:14:36  // probe at 129.513,106.725 is z=0.814725
20:14:37  // probe at 129.513,106.725 is z=0.815975
20:14:39  // probe at 141.750,106.725 is z=0.803427
20:14:40  // probe at 141.750,106.725 is z=0.803427
20:14:41  // probe at 153.975,106.725 is z=0.792129
20:14:42  // probe at 153.975,106.725 is z=0.790879
20:14:42  // Mesh Bed Leveling Complete
20:14:42  // Smart Park location: 95.268,97.816.
20:15:10  // Moving filament tip 1.8mms
20:15:10  // KAMP purge is using firmware retraction.
20:15:10  // KAMP purge starting at 107.5, 97.816 and purging 20.0mm of filament, requested flow rate is 15.0mm3/s.
```

**Reply by phillipthelen** [reply]
Oooh. I think this is the new Klipper "KAMP". the original KAMP does not have the braces. I'll fix it in the next update. This is also tracked in #85

**Reply by phillipthelen** [reply, marked as answer]
The fix is available in the main branch

**Reply by recliq** [reply]
I will update this evening and check!
Thank you for the quick fix!

**Reply by recliq** [reply]
@phillipthelen 
Just to let you know, after updating in fluidd webinterface the touchscreen wasn't working atall...
Looking at the syslog revealed that the service file was missing the display.py file.
I then started `~/OpenNept4une/OpenNept4une.sh` which prompted me for migrating the display repository which I did and the display is now working again.
It seems it's mandatory to run the OpenNept4une installer to migrate the display service which seems to have it's own repository now...

Nevertheless everything is working now!
